modules/tinymce/src/plugins/preview/main/ts/core/IframeContent.ts

```diff
@@ -29,9 +29,17 @@ const getPreviewHtml = (editor: Editor): string => {
   const preventClicksOnLinksScript = (
     '<script>' +
     'document.addEventListener && document.addEventListener("click", function(e) {' +
-    'for (var elm = e.target; elm; elm = elm.parentNode) {' +
-    'if (elm.nodeName === "A" && !(' + isMetaKeyPressed + ')) {' +
+    'for (let elm = e.target; elm; elm = elm.parentNode) {' +
+    'if (elm.nodeName === "A") {' +
+    'const href = elm.getAttribute("href");' +
+    'if (href && href.startsWith("#") && (' + isMetaKeyPressed + ')) {' +
```

Comment:
```suggestion
    'if (href && href.startsWith("#")) {' +
```

I don't remember where I said this, but I think we should scroll whether meta is pressed or not.